### PR TITLE
Search provider improvements

### PR DESCRIPTION
**Pull request by rafaelmardojai**
- [x] Show some info about the search provider  in preferences window
- [x]  Implement `--text` command line option
- [x]  Implement `ActivateResult` on search provider

### CODE REVIEW

**Comment by rafaelmardojai** [comment]
@fushinari I'm going to sleep now. If you have time, feel free to push here.

**Comment by fushinari** [comment]
You'll have to rebase this before continuing work. I wanted to complete this but things came up. Sorry

**Comment by rafaelmardojai** [comment]
Other things to do:

- [ ] Remove result icon
- [ ] Show dest lang name as description

File: data/resources/preferences.ui [hunk 1]
```diff
@@ -59,6 +57,13 @@
             </child>
           </object>
         </child>
+        <child>
+          <object class="HdyPreferencesGroup" id="search_provider">
+            <property name="title" translatable="yes">Search Provider</property>
+            <property name="description" translatable="yes">Dialect has a GNOME Search Provider. For API concerns it requires the Live Translation preference to work. Also be sure it's enabled in GNOME Settings.</property>
```
Comment (rafaelmardojai) [inline data/resources/preferences.ui:63]: This description is fine?

File: src/search_provider/search_provider.in [hunk 2]
```diff
@@ -62,13 +63,14 @@ class TranslateService(dbus.service.Object):
                 pass
 
             name = self.translations[translate_id]
+            lang = LANGUAGES[self.settings.get_strv('dest-langs')[0]].capitalize()
```
Comment (rafaelmardojai) [inline src/search_provider/search_provider.in:66]: Just remembered that the lang not always is from `dest-langs`.

File: data/resources/preferences.ui (hunk 1 above)
Comment (fushinari) [reply data/resources/preferences.ui:63]: image

Is this better or worse? I changed both the descriptions slightly. (Live Translation too)

Comment (rafaelmardojai) [reply data/resources/preferences.ui:63]: I like it.

Comment (fushinari) [reply data/resources/preferences.ui:63]: "Translation Shortcut"
"Unselected choice will be used for line break."

Is that better? I've always felt that preference needs better labeling but I'm very confused how to.

File: src/search_provider/search_provider.in (hunk 2 above)
Comment (fushinari) [reply src/search_provider/search_provider.in:66]: It isn't? How?

File: data/resources/preferences.ui (hunk 1 above)
Comment (rafaelmardojai) [reply data/resources/preferences.ui:63]: At least is better than the former.

File: src/search_provider/search_provider.in (hunk 2 above)
Comment (rafaelmardojai) [reply src/search_provider/search_provider.in:66]: > It isn't? How?

See L158. If the detected and dest langs are the same, it use the first of saved src langs.

**Comment by rafaelmardojai** [comment]
After merging this, what is left to release `1.1.0`?

**Comment by fushinari** [comment]
I have this issue on Plasma where the icon's "D" isn't visible
image

Really odd issue and I don't know if it's an issue with the icon or if it's with Plasma.

**Comment by fushinari** [comment]
Some more testing after these changes and I think we can push 1.1.0.

There's also one more issue. With this complete, we will have two methods of translation, the GUI and the search provider. The issue is that they're not in sync anymore. Changing languages in the app will not change anything in the search provider until the app is closed.

It's not a huge deal but we might want to consider offloading the saving of settings to a non-daemon thread similar to the translation queue but without the cleaning. Then we could save the settings with each change rather than wait for the app to exit.

File: src/search_provider/search_provider.in (hunk 2 above)
Comment (fushinari) [reply src/search_provider/search_provider.in]: Can you check if it works as expected now?

**Comment by rafaelmardojai** [comment]
> I have this issue on Plasma where the icon's "D" isn't visible

I have made a cleanup to the icon. [com.github.gi_lom.dialect.zip](https://github.com/gi-lom/dialect/files/5440237/com.github.gi_lom.dialect.zip)

Try it and let me know.

**Review by fushinari — APPROVED** [review]
> Try it and let me know.

Yep, your icon works fine. Guess the "Devel" icon will also have to be updated.

Gonna sleep now, so you can go ahead and merge when ready. :smile:

**Pull request merged**